vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -169,6 +169,59 @@ ReactInstanceWin::ReactInstanceWin(
 
 ReactInstanceWin::~ReactInstanceWin() noexcept {}
 
+void ReactInstanceWin::LoadModules(
+    const std::shared_ptr<winrt::Microsoft::ReactNative::NativeModulesProvider> &nativeModulesProvider,
+    const std::shared_ptr<winrt::Microsoft::ReactNative::TurboModulesProvider> &turboModulesProvider) noexcept {
+  auto registerNativeModule = [&nativeModulesProvider](
+      const wchar_t *name, const ReactModuleProvider &provider) noexcept {
+    nativeModulesProvider->AddModuleProvider(name, provider);
+  };
+
+  auto registerTurboModule = [ this, &nativeModulesProvider, &turboModulesProvider ](
+      const wchar_t *name, const ReactModuleProvider &provider) noexcept {
+    if (m_options.UseWebDebugger()) {
+      nativeModulesProvider->AddModuleProvider(name, provider);
+    } else {
+      turboModulesProvider->AddModuleProvider(name, provider);
+    }
+  };
+
+  registerTurboModule(L"Alert", winrt::Microsoft::ReactNative::MakeModuleProvider<::Microsoft::ReactNative::Alert>());
```

Comment:
Could you help me understand the matrix:

| Register function | Factory funciton |
|--|--|
|registerTurboModule|MakeModuleProvider|
|registerNativeModule|MakeTurboModuleProvider|
|registerTurboModule|MakeTurboModuleProvider|

Reply:
@dannyvv `MakeTurboModuleProvider` and `MakeModuleProvider` are confusing but it was there for a long time.
- `MakeModuleProvider`: build the module provider
- `MakeTurboModuleProvider`: build the module provider, comparing to a spec class so that C++ will check method signatures

They don't really mean native module and turbo module.

Reply:
thanks!